server/src/main/java/com/cloud/hypervisor/HypervisorGuruBase.java

```diff
@@ -179,9 +179,13 @@ protected VirtualMachineTO toVirtualMachineTO(VirtualMachineProfile vmProfile) {
         ServiceOffering offering = _serviceOfferingDao.findById(vmProfile.getId(), vmProfile.getServiceOfferingId());
         VirtualMachine vm = vmProfile.getVirtualMachine();
         HostVO host = hostDao.findById(vm.getHostId());
+        boolean divideMemoryByOverprovisioning = true;
+        boolean divideCpuByOverprovisioning = true;
 
-        boolean divideMemoryByOverprovisioning = VmMinMemoryEqualsMemoryDividedByMemOverprovisioningFactor.valueIn(host.getClusterId());
-        boolean divideCpuByOverprovisioning = VmMinCpuSpeedEqualsCpuSpeedDividedByCpuOverprovisioningFactor.valueIn(host.getClusterId());
+        if (host != null) {
```

Comment:
nit - should we check both host != null and host.getClusterId != null? (that may not be necessary but for writing defensive code)?

Reply:
host.getClusterId() is not referred, only passed. the check should be up to the called method and the method constructing the host object (i think!)

Reply:
@rhtyd  As @DaanHoogland mentioned, the check for validating if the passed argument is null is done by valueIn() (In ConfigKey class) hence there's no need for a check at this point.